tokio-io/src/split.rs

```diff
@@ -77,6 +110,33 @@ fn wrap_as_io<T>(t: Async<T>) -> Result<Async<T>, io::Error> {
     Ok(t)
 }
 
+/// Error returned from `unsplit` if reuniting a `ReadHalf` with
+/// another `WriteHalf` fails.
+pub struct UnsplitError<T> {
+    /// The `ReadHalf` passed to `unsplit`.
+    pub read_half: ReadHalf<T>,
```

Comment:
This should be private w/ accessors to get it (for forwards compat).

Reply:
I think the primary use case for handling the failure of `unsplit` is to get the two halves back. Accessors do not seem that useful to me here. One would need an impl `Into<(ReadHalf<T>, WriteHalf<T>)> for UnsplitError<T>` in order to get both halves back. If you are concerned about forwards compatibility (and adding a `_priv: ()` field is not good enough), then I would suggest making the fields private, provide no accessors, but the `Into` implementation. However I think this is needlessly involved and adding a `_priv: ()` would be good enough, no?